### DISCUSSION TITLE
Calling getData(evt.pixel) on WMS layer returns null

### DISCUSSION THREAD

**Discussion by samuelchau** [openlayers/openlayers#15744] [Q&A] [answered]
I have a WMS layer on my map and am trying to following this example
https://openlayers.org/en/latest/examples/getfeatureinfo-tile.html
to see if I can request the GetFeatureInfo and pixel data.

The 'singleclick' callback function of the example successfully gets the info from my layer
However, the 'pointermove' callback function returns null from the getData function(i.e.  wmsLayer.getData(evt.pixel); )

Anyone knows why I'm not getting any data from calling getData() function on my WMS layer ?

Any help would be appreciated. 

Here's the definition of my WMS layer I'm using:
 const layerPrecipitation = new TileLayer({
        source: new TileWMS({
          url: "https://view.eumetsat.int/geoserver/wms",
          params: {
            layers: "msg_iodc:h63",
            TILED: true,
            VERSION: "1.3.0",
            TIME:	dateTimeStr,
          }, 
          projection: "EPSG:4326",

        }),
      });

**Comment by ahocevar** [comment]
`getData(pixel)` is not implemented for raster tile and image layers.

**Reply by samuelchau** [reply]
I'm using TileLayer/TileWMS just like the example is using.  
Why wouldn't this work like the example  ?

import TileLayer from 'ol/layer/Tile.js';
import TileWMS from 'ol/source/TileWMS.js';

**Comment by ahocevar** [comment, marked as answer]
Ah sorry, I was on the wrong track. My answer above was referring to `getFeature(pixel)`.

The problem in your code is that your `TileWMS` source is missing the
```js
crossOrigin: 'anonymous',
```
config option. If it still does not work with that setting, then the WMS does not send CORS headers.

**Comment by samuelchau** [comment]
Thanks.  It is working now.
